compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -187,6 +187,10 @@ pub fn predicate_obligations<'tcx>(
         ty::PredicateKind::TypeWellFormedFromEnv(..) => {
             bug!("TypeWellFormedFromEnv is only used for Chalk")
         }
+        ty::PredicateKind::AliasEq(t1, t2) => {
+            wf.compute(t1.into_arg());
+            wf.compute(t2.into_arg());
+        }
```

Comment:
actually, is this unreachable? we should only check wf for where clauses

Reply:
other code here does stuff for `PredicateKind::Subtype` `PredicateKind::Coerce` `PredicateKind::ConstEquate` etc so if this is not supposed to be called on things that arent `Clause` then its definitely not implemented as such :sweat_smile:

Reply:
I guess, `bug!` here for now and we'll figure out what to do once it causes issues in the future